MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/SymbolSwitchOnZoomActivity.java

```diff
@@ -0,0 +1,168 @@
+package com.mapbox.mapboxandroiddemo.examples.dds;
+
+import android.graphics.BitmapFactory;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.Toast;
+
+import com.mapbox.geojson.Feature;
+import com.mapbox.geojson.FeatureCollection;
+import com.mapbox.geojson.Point;
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+
+import static com.mapbox.mapboxsdk.style.expressions.Expression.literal;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.step;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.stop;
+import static com.mapbox.mapboxsdk.style.expressions.Expression.zoom;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconAllowOverlap;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconIgnorePlacement;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconImage;
+
+/**
+ * Use the SymbolLayer's setMinZoom and setMaxZoom methods to create the effect of SymbolLayer icons switching
+ * based on the map camera's zoom level.
+ */
+public class SymbolSwitchOnZoomActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+  private static final float ZOOM_LEVEL_FOR_SWITCH = 12;
+  private static final String BLUE_PERSON_ICON_ID = "blue-car-icon-marker-icon-id";
+  private static final String BLUE_PIN_ICON_ID = "blue-marker-icon-marker-icon-id";
+  private MapView mapView;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_zoom_based_icon_switch);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(final MapboxMap mapboxMap) {
+    mapboxMap.setStyle(Style.OUTDOORS, new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        initLayerIcons(style);
+        addDataToMap(style);
```

Comment:
same for the source/layers? `new StyleBuilder().fromUrl(Streets).withSource(...)`

Reply:
Done for the source. There was no way to `.setProperties` on the `SymbolLayer` _within_ the Style Builder.

For now, I stuck with:

```java
// Create a SymbolLayer and use the {@link com.mapbox.mapboxsdk.style.expressions.Expression.step()}
          // to adjust the SymbolLayer icon based on the zoom level. The blue person icon is set as the default
          // icon and then a step is used to switch to the blue person icon at a certain map camera zoom level.
          SymbolLayer singleLayer = new SymbolLayer("symbol-layer-id", "source-id");
          singleLayer.setProperties(
            iconImage(step(zoom(), literal(BLUE_PERSON_ICON_ID),
              stop(ZOOM_LEVEL_FOR_SWITCH, BLUE_PIN_ICON_ID))),
            iconIgnorePlacement(true),
            iconAllowOverlap(true));
          style.addLayer(singleLayer);
```

If there's a way to do this within `new StyleBuilder()`, then I'm happy to refactor in a new pr.